pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalanceProgressStats.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.controller.helix.core.rebalance;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+
+/**
+ * These are rebalance stats as to how the current state is, when compared to the target state.
+ * Eg: If the current has 4 segments whose replicas (16) don't match the target state, _segmentsToRebalance
+ * is 4 and _replicasToRebalance is 16.
+ */
+public class TableRebalanceProgressStats {
+  public static class RebalanceStateStats {
+    public int _segmentsMissing;
+    public int _segmentsToRebalance;
+    public double _percentSegmentsToRebalance;
+    public int _replicasToRebalance;
+
+    RebalanceStateStats() {
+      _segmentsMissing = 0;
+      _segmentsToRebalance = 0;
+      _replicasToRebalance = 0;
+      _percentSegmentsToRebalance = 0.0;
+    }
+  }
+
+  // Done/In_progress/Failed
+  private String _status;
+  // When did Rebalance start
+  private Long _startTimeInMilliseconds;
```

Comment:
Should we store primitive `long`? I see the caller use the value to calculate the time difference. If it is `null`, it will throw NPE. Also we usually use `Ms` as the shorter name for `InMilliseconds`
```suggestion
  private long _startTimeMs;
```

Reply:
Changed it to long now. I was checking for rebalance status which will ensure that this field is non null (eg: when in_progress, it'll have a startTime).